BGAnimations/ScreenGameplay overlay/FailOnHoldStart.lua

```diff
@@ -1,10 +1,18 @@
--- We need a way to check if the player gave up before the song properly ended.
--- It doesn't look like the engine broadcasts any messages that would be helpful here,
--- so we do the best we can by checking the song position.
+-- Check if the player gave up before the song properly ended.
 
 local af = Def.ActorFrame{
 	OffCommand=function(self)
-		local fail = (GAMESTATE:GetCurMusicSeconds() < GAMESTATE:GetCurrentSong():GetLastSecond())
+		-- In OutFox and newer versions of SM 5.1 there is the GaveUp()
+		-- function available. For SM 5.0.12 and older SM 5.1 versions
+		-- we do the best we can by checking the song position. It
+		-- doesn't look like the engine broadcasts any messages that
+		-- would be helpful here.
+		local stage_stats = STATSMAN:GetCurStageStats()
+		if stage_stats.GaveUp then
+			fail = stage_stats:GaveUp()
```

Comment:
Should `fail` be made local? I always wonder how stuff works without the specifier.

Reply:
Oops, fixed! It's like in javascript... no specifier means it's global :cry: